app.py

```diff
@@ -528,6 +569,40 @@ def _filter_carpools_by_date(leave_time, return_time):
     return pools
 
 
+def _filter_carpools_by_location(from_latitude,
+                                 from_longitude,
+                                 to_latitude,
+                                 to_longitude):
+
+    pools = Carpool.query.filter(
+        Carpool.from_latitude == from_latitude,
+        Carpool.from_longitude == from_longitude,
+        Carpool.to_latitude == to_latitude,
+        Carpool.to_longitude == to_longitude
+    )
+
+    return pools
+
+
+def _get_all_carpool_locations():
+    pools = Carpool.query.all()
```

Comment:
We probably don't want to load *all* the carpool start/end locations onto the map. That could be a lot of things.